// @ballzo-ui/core/lib/es5/utils/file-utility.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dataURLtoFile = exports.sanitizeFileName = exports.FileType = void 0;
exports.FileType = {
    cad: 'image/vnd.dwg',
    excel: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    imageJPEG: 'image/jpeg',
    imagePNG: 'image/png',
    image: 'image/*',
    octetStream: 'application/octet-stream',
    pdf: 'application/pdf',
    word: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
    csv: 'text/csv',
    zip: '.zip',
};
/**
 * Sanitizes the file name
 * @param fileName
 * @returns
 */
function sanitizeFileName(fileName, separator) {
    return fileName.replace(/\s+/g, separator);
}
exports.sanitizeFileName = sanitizeFileName;
/**
 * Converts base64 to file
 * @param dataUrl
 * @param filename
 * @returns {File}
 */
function dataURLtoFile(dataUrl, filename) {
    var arr = dataUrl.split(','), mime = arr[0].match(/:(.*?);/)[1], bstr = atob(arr[arr.length - 1]), n = bstr.length, u8arr = new Uint8Array(n);
    while (n--) {
        u8arr[n] = bstr.charCodeAt(n);
    }
    return new File([u8arr], filename, { type: mime });
}
exports.dataURLtoFile = dataURLtoFile;
