// @ballzo-ui/core/lib/es5/common/common.model.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PositionData = exports.SupportRequest = exports.BackgroundCSS = exports.IUserProperties = exports.IListOption = void 0;
// Classes
var IListOption = /** @class */ (function () {
    function IListOption() {
        this.viewValue = 'Label';
        this.value = null;
    }
    return IListOption;
}());
exports.IListOption = IListOption;
// Interfaces
var IUserProperties = /** @class */ (function () {
    function IUserProperties() {
        this.displayName = "";
        this.photoURL = "";
    }
    return IUserProperties;
}());
exports.IUserProperties = IUserProperties;
var BackgroundCSS = /** @class */ (function () {
    function BackgroundCSS() {
        this.background = '';
        this.color = 'black';
    }
    return BackgroundCSS;
}());
exports.BackgroundCSS = BackgroundCSS;
var SupportRequest = /** @class */ (function () {
    function SupportRequest() {
        this.subject = '';
        this.description = '';
        this.uid = '';
        this.screenshots = [];
        this.createdAt = new Date().getTime();
    }
    return SupportRequest;
}());
exports.SupportRequest = SupportRequest;
/**
 * DO NOT CHANGE THE ORDER OF THE ARRAY VALUES.
 */
exports.PositionData = [
    {
        title: 'Manager',
        subtitle: 'Game Coordinator / Slot Booker',
    },
    { title: 'Striker', subtitle: 'LW / ST / CF / RW' },
    {
        title: 'Midfielder',
        subtitle: 'LM / CM / RM',
    },
    {
        title: 'Defender',
        subtitle: 'LB / CB / RB',
    },
    {
        title: 'Goalkeeper',
        subtitle: 'GK',
    },
];
