// @ballzo-ui/core/lib/es5/common/common-constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabLabel = exports.Regex = exports.SnackbarConstants = exports.LocalStorageProperties = exports.Constants = void 0;
var Constants = /** @class */ (function () {
    function Constants() {
    }
    Constants.GET_STARTED = 'Get Started';
    Constants.SHEET_OPEN_OUTLET = 'open';
    Constants.INPUT_NAME = 'input';
    Constants.NOT_AVAILABLE = '-';
    Constants.loginURL = "(" + Constants.SHEET_OPEN_OUTLET + ":login)";
    Constants.signupURL = "(" + Constants.SHEET_OPEN_OUTLET + ":signup)";
    Constants.RUPEE_SYMBOL = '₹';
    Constants.PLACEHOLDER_ARRAY = [1, 2, 3, 4];
    Constants.DEFAULT_FACILITY_MAX_PLAYER_LIMIT = 22;
    Constants.DEFAULT_FACILITY_ALLOWED_PLAYER_LIMIT = Constants.DEFAULT_FACILITY_MAX_PLAYER_LIMIT;
    Constants.DEFAULT_SLOT_PRICE = 60;
    Constants.FILE_NAME_SEPARATOR = '-';
    Constants.DATE_TIME_FORMATS = {
        format_1: 'dd EEE',
        format_2: 'd MMM, y',
        format_3: 'h:mm a',
        format_4: 'd MMM, hh:mm a',
        format_5: 'EEEE, MMMM d',
        format_6: 'd MMM' // Aug 14
    };
    Constants.NUMBER_FORMATS = {
        format_1: '1.0-0', // 3.6 -> 4
    };
    Constants.SEVEN_MINUTES_IN_MILLISECONDS = 420000;
    Constants.ONE_DAY_IN_MILLISECONDS = 86400000;
    Constants.THREE_DAYS_IN_MILLISECONDS = 259200000;
    Constants.ONE_HOUR_IN_MILLISECONDS = 3600000;
    Constants.TWELVE_HOURS_IN_MILLISECONDS = 43200000;
    Constants.ORDER_PREFIX = 'order';
    Constants.ORDER_PREFIX_UI = 'OID: ';
    Constants.DAY_TIMES = {
        morning: 'Morning',
        afternoon: 'Afternoon',
        evening: 'Evening',
        night: 'Night',
    };
    Constants.DELETED_USER_PLACEHOLDER = 'Unknown User';
    Constants.DEFAULT_IMG = 'assets/images/placeholder.png';
    Constants.PLUS_SIGN = '+';
    Constants.MINUS_SIGN = '-';
    Constants.BOOKING_CANCELLATION_REASONS = [
        'Conflict with another match',
        'Unfavorable weather conditions',
        'Injury or Health Issues',
        'Insufficient Players',
        'Ground/Match Quality',
        'Others'
    ];
    Constants.REASON_PLACEHOLDER = 'Select a reason';
    Constants.LOCATION_API_URL = {
        countryStateCityCommonUrl: 'https://api.countrystatecity.in/v1/countries/IN/states/',
        locationApiHeader: 'X-CSCAPI-KEY',
    };
    Constants.LOGIN_CAPTCHA_PLACEHOLDER = 'recaptcha-container-login';
    Constants.INDIAN_DIAL_CODE = '+91';
    Constants.MAX_DESCRIPTION_LIMIT = 250;
    Constants.MAX_SUBJECT_LIMIT = 50;
    Constants.DEFAULT_PLACEHOLDER = 'Select File';
    Constants.ALLOWED_FORMATS = ['jpg', 'jpeg', 'png'];
    Constants.SUPPORT_REQUEST_ID_PREFIX = 'BA2023';
    Constants.FILE_SEPARATOR = '-';
    Constants.FILE_NAME = 'screenshot-support-';
    Constants.CANCELLATIONS_ALLOWED_IN_WEEK = 3;
    return Constants;
}());
exports.Constants = Constants;
var LocalStorageProperties = /** @class */ (function () {
    function LocalStorageProperties() {
    }
    LocalStorageProperties.BOTTOM_SHEET = 'openSheet';
    LocalStorageProperties.USER_UID = 'uid';
    return LocalStorageProperties;
}());
exports.LocalStorageProperties = LocalStorageProperties;
var SnackbarConstants = /** @class */ (function () {
    function SnackbarConstants() {
    }
    SnackbarConstants.AUTO_HIDE = 3000; // time in milliseconds
    SnackbarConstants.HORIZONTAL_POSITION = 'center'; // time in milliseconds
    SnackbarConstants.VERTICAL_POSITION = 'bottom'; // time in milliseconds
    SnackbarConstants.DEFAULT_ACTION = 'OK';
    return SnackbarConstants;
}());
exports.SnackbarConstants = SnackbarConstants;
var Regex = /** @class */ (function () {
    function Regex() {
    }
    Regex.alphabeticAndSpace = /^[A-Za-z\s]+$/;
    Regex.alphanumeric = /^[A-Za-z0-9]+$/;
    Regex.alphanumericAndSpace = /^[A-Za-z0-9\s]+$/;
    Regex.alphabeticWithNumbersAndSpecialCharacters = /^[A-Za-z0-9\s\.\;\-]+$/;
    Regex.alphanumericWithSpecialCharacters = /^[A-Za-z0-9\.\,\-]+$/;
    Regex.alphanumericWithSpecialCharactersAndSpace = /^[A-Za-z0-9\.\,\-\s]+$/;
    Regex.customNumericMobileNumber = /^[1-9][0-9]{9}$/;
    Regex.customNumericOtp = /^[0-9]{6}$/;
    return Regex;
}());
exports.Regex = Regex;
var TabLabel;
(function (TabLabel) {
    TabLabel[TabLabel["past"] = -1] = "past";
    TabLabel[TabLabel["today"] = 0] = "today";
    TabLabel[TabLabel["tomorrow"] = 1] = "tomorrow";
    TabLabel[TabLabel["dayAfter"] = 2] = "dayAfter";
    TabLabel[TabLabel["farAway"] = 4] = "farAway";
})(TabLabel = exports.TabLabel || (exports.TabLabel = {}));
