// @ballzo-ui/core/lib/es6/common/common.model.js
// Classes
var IListOption = /** @class */ (function () {
    function IListOption() {
        this.viewValue = 'Label';
        this.value = null;
    }
    return IListOption;
}());
export { IListOption };
// Interfaces
var IUserProperties = /** @class */ (function () {
    function IUserProperties() {
        this.displayName = "";
        this.photoURL = "";
    }
    return IUserProperties;
}());
export { IUserProperties };
var BackgroundCSS = /** @class */ (function () {
    function BackgroundCSS() {
        this.background = '';
        this.color = 'black';
    }
    return BackgroundCSS;
}());
export { BackgroundCSS };
var SupportRequest = /** @class */ (function () {
    function SupportRequest() {
        this.subject = '';
        this.description = '';
        this.uid = '';
        this.screenshots = [];
        this.createdAt = new Date().getTime();
    }
    return SupportRequest;
}());
export { SupportRequest };
/**
 * DO NOT CHANGE THE ORDER OF THE ARRAY VALUES.
 */
export var PositionData = [
    {
        title: 'Manager',
        subtitle: 'Game Coordinator / Slot Booker',
    },
    { title: 'Striker', subtitle: 'LW / ST / CF / RW' },
    {
        title: 'Midfielder',
        subtitle: 'LM / CM / RM',
    },
    {
        title: 'Defender',
        subtitle: 'LB / CB / RB',
    },
    {
        title: 'Goalkeeper',
        subtitle: 'GK',
    },
];
