// @ballzo-ui/core/lib/es5/transaction/order.model.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CancellationBehavior = exports.OrderRz = exports.OrderReturn = void 0;
var OrderReturn = /** @class */ (function () {
    function OrderReturn() {
        this.id = '';
        this.reason = '';
        this.oid = '';
        this.returnCount = 0;
        this.timestamp = new Date().getTime();
        this.returnMode = 'ballzo-wallet';
    }
    return OrderReturn;
}());
exports.OrderReturn = OrderReturn;
var OrderRz = /** @class */ (function () {
    function OrderRz() {
        this.id = '';
        this.bookingId = '';
        this.uid = '';
        this.entity = '';
        this.amount = 0;
        this.amount_paid = 0;
        this.amount_due = 0;
        this.currency = 'INR';
        this.receipt = '';
        this.offer_id = '';
        this.offers = {};
        this.status = 'created';
        this.attempts = 0;
        this.notes = {};
        this.created_at = new Date().getTime();
    }
    Object.defineProperty(OrderRz.prototype, "_amount", {
        get: function () {
            return Number(this.amount) / 100;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(OrderRz.prototype, "_created_at", {
        get: function () {
            return this.created_at * 1000;
        },
        enumerable: false,
        configurable: true
    });
    return OrderRz;
}());
exports.OrderRz = OrderRz;
var CancellationBehavior = /** @class */ (function () {
    function CancellationBehavior() {
        this.lastCancellationTimestamp = 0;
        this.count = 0;
    }
    CancellationBehavior.prototype._updateCancellationCount = function () {
        this.count++;
    };
    CancellationBehavior.prototype._resetCount = function () {
        this.count = 0;
    };
    return CancellationBehavior;
}());
exports.CancellationBehavior = CancellationBehavior;
