// @ballzo-ui/core/lib/es5/user/user.model.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserWallet = exports.PlayerStats = exports.Player = exports.Position = void 0;
var common_constants_1 = require("../common/common-constants");
/**
 * Defines the position of a player.
 * DO NOT CHANGE THE ORDER OF THE ENUM VALUES.
 */
var Position;
(function (Position) {
    Position["manager"] = "manager";
    Position["striker"] = "striker";
    Position["midfielder"] = "midfielder";
    Position["defender"] = "defender";
    Position["goalkeeper"] = "goalkeeper";
})(Position = exports.Position || (exports.Position = {}));
var Player = /** @class */ (function () {
    function Player() {
        this.id = '';
        this.name = '';
        this.imgLink = '';
        this.position = Position.striker;
        this.dob = 0;
        this.locationCity = '';
        this.locationState = '';
    }
    Object.defineProperty(Player.prototype, "_location", {
        /**
         * Gets the location
         * @returns {string}
         */
        get: function () {
            if (this.locationCity && this.locationState) {
                return this.locationCity + ", " + this.locationState;
            }
            else if (this.locationCity) {
                return this.locationCity;
            }
            else if (this.locationState) {
                return this.locationState;
            }
            else {
                return common_constants_1.Constants.NOT_AVAILABLE;
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(Player.prototype, "_dateOfBirthString", {
        /**
         * Gets the date of birth
         */
        get: function () {
            if (this.dob) {
                return new Date(this.dob);
            }
            return null;
        },
        enumerable: false,
        configurable: true
    });
    return Player;
}());
exports.Player = Player;
var PlayerStats = /** @class */ (function () {
    function PlayerStats() {
        this.apps = 0;
        this.goals = 0;
        this.wins = 0;
    }
    return PlayerStats;
}());
exports.PlayerStats = PlayerStats;
var UserWallet = /** @class */ (function () {
    function UserWallet() {
        this.amount = 0;
        this.disabled = false;
    }
    return UserWallet;
}());
exports.UserWallet = UserWallet;
